src/auth/mod.ts

```diff
@@ -15,33 +14,131 @@ interface DecodedToken {
 
 const logger = getLogger(import.meta);
 
-function isSubInPodsConfig(payload: DecodedToken): boolean {
-  return podsConfig.pods.some((pod) =>
-    pod.sub === payload.sub &&
-    pod.serviceaccount.uid === payload["kubernetes.io"]?.serviceaccount.uid
+export async function authenicateRequestAndReturnPodServiceAccount(
```

Comment:
does that function really "authenticate" ?

Reply:
aren't we basically authenticating the request to access herald if the token it carries is valid?